File: backend/app/schemas/life_area.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class LifeAreaResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str
    slug: str
    description: str | None
    icon: str | None
    is_default: bool
    created_at: datetime


class UserAreaResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    user_id: int
    life_area_id: int
    is_active: bool
    created_at: datetime


class CustomLifeAreaCreate(BaseModel):
    name: str = Field(min_length=2, max_length=100)
    description: str | None = Field(default=None, max_length=500)
    icon: str | None = Field(default=None, max_length=50)

File: frontend/src/types/lifeArea.ts
export type LifeAreaResponse = {
  id: number;
  name: string;
  slug: string;
  description: string | null;
  icon: string | null;
  is_default: boolean;
  created_at: string;
};

export type UserAreaResponse = {
  id: number;
  user_id: number;
  life_area_id: number;
  is_active: boolean;
  created_at: string;
};